### PR TITLE
HHH-18737 Introduce util methods for get bean from BeanContainer

### PR DESCRIPTION
**Pull request by quaff**
[Please describe here what your change is about]

----------------------
By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt)
and can be relicensed under the terms of the [LGPL v2.1 license](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.txt) in the future at the maintainers' discretion.
For more information on licensing, please check [here](https://github.com/hibernate/hibernate-orm/blob/main/CONTRIBUTING.md#legal).

----------------------

https://hibernate.atlassian.net/browse/HHH-18737

### CODE REVIEW

**Comment by quaff** [comment]
Ready for review @gavinking

**Comment by gavinking** [comment]
> Thanks @quaff, appreciated.

Also appreciate your patience on all this.

**Comment by quaff** [comment]
> > Thanks @quaff, appreciated.
> 
> Also appreciate your patience on all this.

It’s my honor to participate in this great project.

File: hibernate-core/src/main/java/org/hibernate/boot/model/internal/GeneratorBinder.java [hunk 1]
```diff
@@ -498,35 +496,19 @@ private static Generator instantiateGeneratorAsBean(
...
+			false,
+			true,
+			() -> instantiateGenerator(
+				annotation,
+				memberDetails,
+				annotationType,
+				creationContext,
+				generatorClass
+				)
+			);
```
Comment (gavinking) [inline hibernate-core/src/main/java/org/hibernate/boot/model/internal/GeneratorBinder.java:511]:
```suggestion
			)
		);
```

**Comment by gavinking** [comment]
Thanks again.

**Pull request merged**